tests/checks/abbr.fish

```diff
@@ -141,7 +141,7 @@ echo $status
 abbr --query banana --function
 echo $status
 # CHECKERR: abbr: --function: option requires an argument
-# CHECKERR: checks/abbr.fish (line 141):
+# CHECKERR: {{.*}}checks/abbr.fish (line 141):
```

Comment:
We have two directories here:

- The "test root", which is in test/ in the build directory when ran via cmake, and just wherever you run the tests from otherwise
- The "homedir", which is a temporary directory (/tmp/randomstring)

What happened before was that test_driver would cd to the homedir, but then the test runner (test.fish or interactive.fish) would `cd (status dirname)` back to the test root.

I removed that last part because that just undid the former.

Now, ending up in the test root means that we get the nice "checks/foo.fish", but it also means that the running test is in a non-temporary directory. Which isn't a huge deal when running with cmake because the test root will also be temporary (it copies the tests there), but outside of cmake it's just your $PWD.

So the test would have to know to `cd` to $HOME if it needs to create files instead. I went with the solution that leaves the $PWD clean by default.

Reply:
Okay, this sent me on a bit of a hike.

The multiple layers of posix shell here make this an absolute pain in the gluteals

(seriously, *how* do you source a script *while* keeping your arguments intact without that script getting them?)

but I did manage.